ui/appui-layout-react/src/appui-layout-react/widget/TabBar.tsx

```diff
@@ -61,25 +61,11 @@ export const WidgetTabBar = React.memo(function WidgetTabBar(props: WidgetTabBar
 
   const containerRef = React.useRef<HTMLDivElement>(null);
   const handleDragStart = React.useCallback((initialPointerPosition: Point, pointerPosition: Point) => {
-    /* if floating widget extract the bounding rect and update state in case bounds are set by content.
-      This is needed so the drag operation can keep widget inside ninezone container. */
-    if (floatingWidgetId) {
-      // istanbul ignore next
-      const containerRect = containerRef.current?.closest(".nz-widget-floatingWidget")?.getBoundingClientRect();
-      // istanbul ignore else
-      if (containerRect) {
-        dispatch({
-          type: "FLOATING_WIDGET_SET_BOUNDS",
-          id: floatingWidgetId,
-          bounds: { left: containerRect.left, top: containerRect.top, right: containerRect.right, bottom: containerRect.bottom },
-        });
-      }
-    }
```

Comment:
Removing that snippet causes an issue where the code that "snaps" back the full widget into view when dragged partially out of the viewport seems to be using the docked size of the widget rather than its currently displayed size if the widget have not been manually resized after being floated. (For example, if I resize the widget after floating it, then the size will be correctly used)

https://user-images.githubusercontent.com/1904889/205135619-fd94905e-f622-4ff9-a323-cff30e9ada7e.mp4

Reply:
Added a test for this case and fixed.

Reply:
Working fine!